### PR TITLE
Fix `FileWatcher` `DELETE` event reporting and rearrange tests

### PR DESCRIPTION
**Pull request by leandro-lucarella-frequenz**
This pull request have 2 different but related main changes:

* The `FileWatcher` was skipping `DELETE` events, except on very weird cases where the file was deleted and re-created just before the event was triggered by `awatch()`.

  This also extends the integration test to test both situations (delete events when a file exist and when it doesn't).

  Because there is usually a considerable delay between a change in the filesystem happens and `awatch()` emits an event it is difficult to decide if we should filter events based on the file existing or not. For example, if a create event is received but before is triggered the file was removed, should we emit the create event or not? Or the other way around, should we emit a delete event if a file was created again before the event was triggered.

  It is probably best to leave this decision to the user, and let them deal with races.

* `FileWatcher` tests were rearranged and split between unit and integration tests.

  The existing tests were really integration tests, as they are not only testing `FileWatcher`, but also `Select` and `PeriodicTimer`, which is confusing when there is a bug in any of the other classes as we might get test failures here (this actually happened during the development of #79).
    
  Because of this the current tests were moved to `test/util/test_integration.py`. We also mark the tests with the `pytest` mark `integration` to make it more clear they are integration tests, and to enable an easy way to only run (exclude) integration tests.

  New unit tests were added to the current test file, properly mocking `awatch` to avoid interactions with the OS.

  There are many more integration tests posing as unit tests, but this PR only takes care of `FileWatcher`. With time we should move the rest too.

### CODE REVIEW

**Comment by leandro-lucarella-frequenz** [comment]
Rebased and added a note on how to test the new timer with `async-solipsism` now.

I would make a release after this is merged.

File: RELEASE_NOTES.md
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md]: I think it should be `either` instead of `neither`, otherwise the sentence is double negative `we don't need...neither`

```suggestion
  Also the new `Timer` uses the `asyncio` loop monotonic clock and the old one used the wall clock (`datetime.now()`) to track time. This means that when using `async-solipsism` to test, the new `Timer` will always trigger immediately regarless of the state of the wall clock.  This also means that we don't need to mock the wall clock with `time-machine` either now.
```

File: RELEASE_NOTES.md
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md:2]: The commit  `Fix FileWatcher skipping DELETE events` has comments that you might want to remove

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I have only 2 cosmetic comments to check for, LGTM otherwise

**Comment by daniel-zullo-frequenz** [comment]
> I would make a release after this is merged.

@leandro-lucarella-frequenz can you please wait for a simple PR that addresses #53 which is based on the changes on this PR?

File: RELEASE_NOTES.md
Comment (leandro-lucarella-frequenz) [reply RELEASE_NOTES.md:2]: Done

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**
